lib/dialogs/share/index.jsx

```diff
@@ -116,12 +119,13 @@ export class ShareDialog extends Component {
                   onSubmit={this.onAddCollaborator}
                 >
                   <input
-                    ref={e => (this.collaboratorElement = e)}
-                    type="email"
-                    pattern="[^@]+@[^@]+"
                     className="settings-item-text-input transparent-input"
+                    pattern="^([^\x00-\x20\x22\x28\x29\x2c\x2e\x3a-\x3c\x3e\x40\x5b-\x5d\x7f-\xff]+|\x22([^\x0d\x22\x5c\x80-\xff]|\x5c[\x00-\x7f])*\x22)(\x2e([^\x00-\x20\x22\x28\x29\x2c\x2e\x3a-\x3c\x3e\x40\x5b-\x5d\x7f-\xff]+|\x22([^\x0d\x22\x5c\x80-\xff]|\x5c[\x00-\x7f])*\x22))*\x40([^\x00-\x20\x22\x28\x29\x2c\x2e\x3a-\x3c\x3e\x40\x5b-\x5d\x7f-\xff]+|\x5b([^\x0d\x5b-\x5d\x80-\xff]|\x5c[\x00-\x7f])*\x5d)(\x2e([^\x00-\x20\x22\x28\x29\x2c\x2e\x3a-\x3c\x3e\x40\x5b-\x5d\x7f-\xff]+|\x5b([^\x0d\x5b-\x5d\x80-\xff]|\x5c[\x00-\x7f])*\x5d))*(\.\w{2,})+$"
```

Comment:
Wow this is quite the regex 😅 

Can we add a comment summarizing it in English?

I found this on a web search as well: https://emailregex.com/

We could probably stand to make sure we're standardizing this across clients.

Reply:
what's our purpose in adding this here? that's probably the heart of the issue. email validators are notoriously flakey and likely to cause more headache than help.

if we want to prevent accidental submission then we might want to consider the most lenient of matchers…

```html
<input pattern="^[^@]+@.+" />
```

Because the server will confirm if it's valid or not. Just leaving some thoughts…

Reply:
So yeah I really more meant "can we add a comment that breaks down exactly what this regex is doing and where it came from". Is it a copy of something from the backend code? Or the same as what's used on the other clients? If not I would prefer we either use one of those instead of introducing something subtly different. Ideally this should be the same validation across the board.

Reply:
I've seen these mega email RegExps many times and they usually cause more bugs than they resolve. The backend doesn't do what you might expect it to, and most don't because usually it's much more productive to resolve an issue when someone has used the wrong or invalid email address than it is to prevent them from doing so.

Reply:
The backed does not do any validation for emails.  If it is not an email it adds it as a tag. We have to have an email regex o this field.

Reply:
> We have to have an email regex o this field.

Sure. I'm only suggesting that we relax the validation to the most basic form to prevent dealing with the validator refusing a bunch of legitimate emails _or_ having to maintain the pattern that nobody can read.

At a minimum, I concur with @codebykat that this needs an explanatory comment.

Reply:
I have switched to a simpler regex.  That regex will ensure `blah@blah.blah`. This regex will not ensure valid emails. If an invalid email is added to the field it will not be added as a collaborator, it will be added as a tag. We can revisit in another PR, I waa get this one merged.

Reply:
>  This regex will not ensure valid emails.

Part of where I'm coming from is that no Regex will. Even if we get what looks like a valid email address it may have no inbox on the other side.

I went looking on the server and found [another insufficient RegExp](https://github.com/Simperium/simplenote-gae/commit/670aa58604d24cf185643391f6a192681755d126) 🙃 and already we can see that it's going to block domains like `.coffee`

:sigh:

Anyway this is why I like relaxed email validation 🤷‍♂ 

> We can revisit in another PR, I want get this one merged.

Just noting that I think this hits that longstanding issue where you don't know if you added the right email address and if your collaborator actually was able to open the note.